routes/compat/wakatime/v1/leaders.go

```diff
@@ -0,0 +1,81 @@
+package v1
+
+import (
+	"github.com/go-chi/chi/v5"
+	"github.com/muety/wakapi/helpers"
+	"github.com/muety/wakapi/middlewares"
+	"github.com/muety/wakapi/models"
+	routeutils "github.com/muety/wakapi/routes/utils"
+	"github.com/muety/wakapi/utils"
+	"net/http"
+
+	conf "github.com/muety/wakapi/config"
+	v1 "github.com/muety/wakapi/models/compat/wakatime/v1"
+	"github.com/muety/wakapi/services"
+)
+
+type LeadersHandler struct {
+	config          *conf.Config
+	userSrvc        services.IUserService
+	leaderboardSrvc services.ILeaderboardService
+}
+
+func NewLeadersHandler(userService services.IUserService, leaderboardService services.ILeaderboardService) *LeadersHandler {
+	return &LeadersHandler{
+		userSrvc:        userService,
+		leaderboardSrvc: leaderboardService,
+		config:          conf.Get(),
+	}
+}
+
+func (h *LeadersHandler) RegisterRoutes(router chi.Router) {
+	router.Group(func(r chi.Router) {
+		r.Use(middlewares.NewAuthenticateMiddleware(h.userSrvc).Handler)
+		r.Get("/compat/wakatime/v1/leaders", h.Get)
+	})
+}
+
+// @Summary List of users ranked by coding activity in descending order.
+// @Description Mimics https://wakatime.com/developers#leaders
+// @ID get-wakatime-user
+// @Tags wakatime
+// @Produce json
+// @Param user path string true "User ID to fetch (or 'current')"
+// @Security ApiKeyAuth
+// @Success 200 {object} v1.UserViewModel
+// @Router /compat/wakatime/v1/leaders [get]
+func (h *LeadersHandler) Get(w http.ResponseWriter, r *http.Request) {
+	wakapiUser, err := routeutils.CheckEffectiveUser(w, r, h.userSrvc, "current")
+	if err != nil {
+		return // response was already sent by util function
+	}
+	pageParams := utils.ParsePageParamsWithDefault(r, 1, 100)
+
+	/*
+		language - String - optional - Filter leaders by a specific language.
+		is_hireable - Boolean - optional - Filter leaders by the hireable badge.
+		country_code - String - optional - Filter leaders by two-character country code.
+		page - Integer - optional - Page number of leaderboard. If authenticated, defaults to the page containing the currently authenticated user.
+	*/
+
+	by := models.SummaryLanguage
+	leaderboard, err := h.leaderboardSrvc.GetAggregatedByInterval(models.IntervalPast7Days, &by, pageParams, true)
+	if err != nil {
+		conf.Log().Request(r).Error("error while fetching general leaderboard items - %v", err)
+		helpers.RespondJSON(w, r, http.StatusInternalServerError, nil)
+		return
+	}
+
+	// regardless of page, always show own rank
+	if wakapiUser != nil && !leaderboard.HasUser(wakapiUser.ID) {
+		// but only if leaderboard spans multiple pages
+		if count, err := h.leaderboardSrvc.CountUsers(); err == nil && count > int64(pageParams.PageSize) {
+			if l, err := h.leaderboardSrvc.GetByIntervalAndUser(models.IntervalPast7Days, wakapiUser.ID, true); err == nil && len(l) > 0 {
+				leaderboard = append(leaderboard, l[0])
+			}
+		}
+	}
+
+	user := v1.NewLeaderboardViewModel(&leaderboard, wakapiUser)
```

Comment:
Code does not compile for me.

```
routes/compat/wakatime/v1/leaders.go:79:13: undefined: v1.NewLeaderboardViewModel
```

Looks like you forgot to commit that file?

Reply:
Yep. Since the previous pull-request overwrote my whole changes I lost the file. Thank god for Goland's "Local History" I got it now back :)